### PR TITLE
Parse and execute multiple statements at once

### PR DESCRIPTION
**Pull request by piotrkira**
It allows to execute multiple statements and also prevents sending empty statement to execution which resulted in unhandled error.

---
It is just a simple for loop, we could move it's logic to separate function, what do you think?

### CODE REVIEW

**Comment by psarna** [comment]
Thanks! The logic has to be a little more complex, because semicolons can also be embedded in strings. In order to support all kinds of multiple statements, we'd have to essentially parse the input first, so that we only split on semicolons that are actually delimiters.

Counterexample that would break with a simple split+loop:
```sql
select ';';
```

**Comment by piotrkira** [comment]
> Thanks! The logic has to be a little more complex, because semicolons can also be embedded in strings. In order to support all kinds of multiple statements, we'd have to essentially parse the input first, so that we only split on semicolons that are actually delimiters.
> 
> Counterexample that would break with a simple split+loop:
> 
> ```sql
> select ';';
> ```

Good point. I've made some changes that should fix that. Anyway, it should recognize any embedded semicolon within single quotes. I guess I also should escape double quotes as one  could create table name that contains semicolon. But for now I would like to know what do you think about such solution?

File: src/rust/libsql-shell/src/main.rs [hunk 1]
```diff
@@ -38,6 +38,40 @@ fn execute(stmt: &mut Statement) -> Result<Vec<Vec<String>>> {
...
+                continue;
+            }
+            if embedded || char != ';' {
+                continue;
+            }
+            let str_statement = self.value[pos..index+1].to_string();
+            if str_statement.starts_with(';') || str_statement.is_empty() {
+                pos = index+1;
+                continue;
+            }
```
Comment (piotrkira) [inline src/rust/libsql-shell/src/main.rs:63]: We could remove that here and filter iterator results but honestly idk which option is better.

**Comment by psarna** [comment]
Sweet, thanks for contributing! 🫡

**Pull request merged**